src/modules/Dropdown/Dropdown.js

```diff
@@ -768,6 +773,9 @@ export default class Dropdown extends Component {
 
     // filter by search query
     if (search && searchQuery) {
+      // filter any component elements like Header or Divider
+      filteredOptions = _.filter(filteredOptions, opt => !opt.component)
```

Comment:
Oh, this is pretty simple.  Perhaps we should just keep `options` as the prop for this case.  For some reason, I hadn't thought of this simple fix.  I believe this will work in all cases.